flexget/plugins/input/dynamic_imdb.py

```diff
@@ -112,130 +138,126 @@ def get_entity_type_and_object(self, imdb_id):
             log.info('Starting to retrieve items for Character: %s' % character)
             return 'Character', character
 
-    def get_items_by_person(self, person, job_types, content_types):
+    def get_items_by_entity(self, entity_type, entity_object, content_types, job_types):
         """
-        Return item list for a person object
-        :param person: Person object
-        :param job_types: Relevant job types as inputted in config
-        :param content_types: Relevant content types as inputted in config
-        :return: Person item list
+        Gets entity object and return movie list using relevant method
         """
-        items = []
-        for job_type in job_types:
-            if job_type in JOBS_WITHOUT_CONTENT_TYPE:
-                log.debug('Searching for movies that correlates to: ' + job_type)
-                items += person.get(job_type)
-                continue
+        if entity_type == 'Company':
+            return self.items_by_company(entity_object)
+
+        if entity_type == 'Character':
+            return self.items_by_character(entity_object, content_types)
+
+        elif entity_type == 'Person':
+            return self.items_by_person(entity_object, job_types, content_types)
+
+    def flatten_list(self, _list):
```

Comment:
Trying to switch to using chain turns out to be a real pain and requires some more changes. If it's not a big deal, I'd like to remain with the current method :)